qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -52,43 +75,145 @@ def _default_options(self):
+        """
+        Produce a list in the size of num_qubits. Each entry value produced from normal distribution with expected value
+        of '0' and standard deviation of self._iq_cluster_width.
+        Args:
+            num_qubits(int): The amount of qubits in the circuit.
 
-            if state > 0.5:
-                point_i = self._iq_cluster_centers[0] + rand_i[idx]
-                point_q = self._iq_cluster_centers[1] + rand_q[idx]
+        Returns:
+            List[float]: A list with values that were produced from normal distribution.
+        """
+        widths = self._iq_cluster_width
+        samples = [self._rng.normal(0, widths[qubit], size=1) for qubit in range(num_qubits)]
+        return samples
+
+    def _probability_dict_to_probability_array(self, prob_dict: Dict[str, float], num_qubits: int) -> List[float]:
+        prob_list = []
+        for num in range(2 ** num_qubits):
+            num_binary_str = str(format(num, 'b').zfill(num_qubits))
+            if num_binary_str in prob_dict:
+                prob_list.append(prob_dict[num_binary_str])
             else:
-                point_i = self._iq_cluster_centers[2] + rand_i[idx]
-                point_q = self._iq_cluster_centers[3] + rand_q[idx]
-
-            if not np.allclose(phase, 0.0):
-                complex_iq = (point_i + 1.0j * point_q) * np.exp(1.0j * phase)
-                point_i, point_q = np.real(complex_iq), np.imag(complex_iq)
+                prob_list.append(0)
+        return prob_list
 
-            memory.append([[point_i, point_q]])
+    def _draw_iq_shots(
+        self, prob: List[float], shots: int, num_qubits: int, phase: float = 0.0
+    ) -> List[List[List[Union[float, complex]]]]:
+        """
+        Produce an IQ shot.
+        Args:
+            prob(List): A list of probabilities for each output.
+            shots(int): The number of times the circuit will run.
+            num_qubits(int): The number of qubit in hte circuit.
 
-        return memory
+        Returns:
+            List[List[Tuple[float, float]]]: return a list that each entry is a list that represent a shot.
+            The output structure is  - List[shot index][qubit index] = [I,Q]
+        """
+        # Randomize samples
+        qubits_iq_rand = []
+        for _ in range(shots):
+            rand_i = np.squeeze(np.array(self._get_normal_samples_for_shot(num_qubits)))
+            rand_q = np.squeeze(np.array(self._get_normal_samples_for_shot(num_qubits)))
+            qubits_iq_rand.append(np.array([rand_i, rand_q], dtype="float").T)
 
-    @abstractmethod
-    def _compute_probability(self, circuit: QuantumCircuit) -> float:
-        """Compute the probability used in the binomial distribution creating the IQ shot.
+        memory = []
+        shot_num = 0
+        iq_centers = self._iq_cluster_centers
+
+        for output_number, number_of_occurrences in enumerate(self._rng.multinomial(shots, prob, size=1)[0]):
```

Comment:
1. As we talked, we don;t know if the perfomance issue is important in this case. Maybe it's better to push it to a new PR.
2. The multinomial function works with an array as input ([documantation].(https://numpy.org/doc/stable/reference/random/generated/numpy.random.multinomial.html))
I am using dictionaries as I think it will be more comfortable and readable than using arrays.